railties/test/commands/temp_test.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require "isolation/abstract_unit"
+require "rails/command"
+
+class Rails::Command::TempTest < ActiveSupport::TestCase
+  setup :build_app
+  teardown :teardown_app
+
+  test "tmp:clear clear cache, socket, screenshot, and storage files" do
+    Dir.chdir(app_path) do
+      FileUtils.mkdir_p("tmp/cache")
+      FileUtils.touch("tmp/cache/cache_file")
+
+      FileUtils.mkdir_p("tmp/sockets")
+      FileUtils.touch("tmp/sockets/socket_file")
+
+      FileUtils.mkdir_p("tmp/screenshots")
+      FileUtils.touch("tmp/screenshots/fail.png")
+
+      FileUtils.mkdir_p("tmp/storage/6h/np")
+      FileUtils.touch("tmp/storage/6h/np/6hnp81jvgt42pcfqtlpoy8qshfb0")
+
+      rails "tmp:clear"
+
+      assert_not File.exist?("tmp/cache/cache_file")
+      assert_not File.exist?("tmp/sockets/socket_file")
+      assert_not File.exist?("tmp/screenshots/fail.png")
+      assert_not File.exist?("tmp/storage/6h/np/6hnp81jvgt42pcfqtlpoy8qshfb0")
```

Comment:
It might be worth adding a private `assert_file` helper for these that also incorporates `app_path` calls.  Then we can drop the `Dir.chdir` block.

Reply:
I've added an `assert_dir_cleared` that checks the content of the directory.
To make sure we don't delete the directory, this assertion also fails if the directory no longer exists.